conan/tools/env/environment.py

```diff
@@ -224,6 +225,24 @@ def compose(self, other):
                     self._values[k] = new_value
         return self
 
+    def to_dict(self):
+        ret = OrderedDict()
+        for varname, varvalues in self._values.items():
+            value = self._format_value(varname, varvalues, "", os.pathsep)
```

Comment:
Beware, this might have issues, you are removing placeholder for previous values of the env-var, like ``"{name}"`` default argument.  Every platform will need to define its one placeholder like ``%{name}%`` in Windows.

This might also do badly for subsystems in which the ``os.pathsep`` is not the system one (msys, cygwin, etc)

Reply:
🤔 I would need to think about it but at first sight, I'm not sure that a value with a placeholder in a python variable makes any sense for a user willing to have a value. That only makes sense in a shell script. right? should be emulate that replacement of the placeholder reading the actual `os.environ`?

Reply:
Yep, exactly, I guess this should be handled in-memory, resolving against the current environment.

Reply:
updated